### PR TITLE
Fix validate issue

### PR DESCRIPTION
**Pull request by n063h**
Fixes  ISSUE#xxx

Changes:

1. Provide correct error msg if chart's value template example not found
2. Set 2 types of errors, so that we can decide which to skip validation
3. checking mapping between the new backend configuration and modules

before
```
❯ kubefate cluster install -f examples/test.yaml
Config Warning: invalid character 'p' after top-level value
create job Success, job id=9fba10be-d757-4c67-bff3-4dd9404439c7
```

after
```
❯ ./bin/kubefate cluster install -f examples/test.yaml
create job Success, job id=fd993d7c-e5a4-4e50-9fad-1be50a8d6bf8
```

### CODE REVIEW

**Comment by JingChen23** [comment]
@n063h You need to build up the habit of pasting the "before fix", "after fix" section in your pr.

Will I get a "get value template example for validation failed\n %v, you may need to upgrade KubeFATE service" msg after your change? If so, then we need to hide this information.

File: k8s-deploy/pkg/cli/validation.go [hunk 1]
```diff
@@ -316,65 +324,250 @@ func getSkippedKeys(m map[string]interface{}) (skippedKeys []string) {
...
+	for _, c := range common {
+		if !Contains(c, modules) {
+			errs = append(errs, ConfigError(fmt.Sprintf("common module %s not enabled", c)))
+		}
+	}
+	return
+}
+
+// checkModuleBackend checks each module has correct backend mapping
+func checkModuleBackend(modules []string, backend map[string]string) (errs []error) {
```
Comment (JingChen23) [inline k8s-deploy/pkg/cli/validation.go:374]: This function LGTM

File: k8s-deploy/pkg/cli/cluster_cli.go [hunk 2]
```diff
@@ -162,14 +162,13 @@ func ClusterInstallCommand() *cli.Command {
...
-					color.Yellow("Config Warning: %s\n", err.Error())
-				}
-			} else {
-				skippedKeys := getSkippedKeys(m)
-				errs := ValidateYaml(template, string(clusterConfig), skippedKeys)
+
+			var errs []error
+			template, err := GetValueTemplateExample(chartName.(string), chartVersion.(string))
+			errs = append(errs, err)
+			skippedKeys := append(getSkippedKeys(m), "route_table")
```
Comment (JingChen23) [inline k8s-deploy/pkg/cli/cluster_cli.go:169]: Cool

**Review by JingChen23 — APPROVED** [review]

File: k8s-deploy/pkg/cli/cluster_cli.go [hunk 3]
```diff
@@ -162,14 +162,13 @@ func ClusterInstallCommand() *cli.Command {
...
+			errs = append(errs, ValidateYaml(template, string(clusterConfig), skippedKeys, nil)...)
+			if !ContainsSkipError(errs) {
 				for _, err := range errs {
 					if err != nil {
 						color.Yellow("Config Warning: %s\n", err.Error())
```
Comment (owlet42) [inline k8s-deploy/pkg/cli/cluster_cli.go:174]: https://github.com/FederatedAI/KubeFATE/blob/master/k8s-deploy/kubefate.yaml#L64-L65

Can this configuration control the log with or without color?

**Pull request merged**